testing/go-ech/ech.go

```diff
@@ -0,0 +1,128 @@
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ech
+
+import (
+	"bytes"
+	"context"
+	"debug/buildinfo"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/go-elasticsearch/v8"
+)
+
+// VerifyEnvVars ensures that the env vars to connect to ES are set, and that the ES_HOST starts with https
+func VerifyEnvVars(t *testing.T) {
+	t.Helper()
+	esHost := os.Getenv("ES_HOST")
+	require.NotEmpty(t, esHost, "Expected env var ES_HOST to be not-empty.")
+	require.Regexp(t, regexp.MustCompile(`^https://`), esHost)
+	esUser := os.Getenv("ES_USER")
+	require.NotEmpty(t, esUser, "Expected env var ES_USER to be not-empty.")
+	esPass := os.Getenv("ES_PASS")
+	require.NotEmpty(t, esPass, "Expected env var ES_PASS to be not-empty.")
+}
+
+// VerifyFIPSBinary ensures the binary on binaryPath has FIPS indicators.
+func VerifyFIPSBinary(t *testing.T, binaryPath string) {
+	t.Helper()
+	info, err := buildinfo.ReadFile(binaryPath)
+	require.NoError(t, err)
+
+	var checkLinks, foundTags, foundExperiment bool
+	for _, setting := range info.Settings {
+		switch setting.Key {
+		case "-tags":
+			foundTags = true
+			require.Contains(t, setting.Value, "requirefips")
+			require.Contains(t, setting.Value, "ms_tls13kdf")
+			continue
+		case "GOEXPERIMENT":
+			foundExperiment = true
+			require.Contains(t, setting.Value, "systemcrypto")
+			continue
+		case "-ldflags":
+			if !strings.Contains(setting.Value, "-s") {
+				checkLinks = true
+			}
+		}
+	}
+
+	require.True(t, foundTags, "did not find build tags")
+	require.True(t, foundExperiment, "did not find GOEXPERIMENT")
+
+	if checkLinks && runtime.GOOS == "linux" {
+		t.Log("Binary is not stripped, checking for OpenSSL in the symbols table.")
+		output, err := exec.CommandContext(t.Context(), "go", "tool", "nm", binaryPath).Output()
+		require.NoError(t, err, "unable to run go tool nm")
+		require.Contains(t, output, "OpenSSL_version", "Unable to find OpenSSL_version in symbols link")
+	}
+}
+
+// RunSmokeTest runs the beat on binaryPath with the passed config, and ensures that data ends up in Elasticsearch.
+func RunSmokeTest(t *testing.T, name, binaryPath, cfg string) {
```

Comment:
[Blocker]
Why not use the test framework we already have? I believe it's better to avoid creating new ways to kinda do the same thing.

I get you need to run a standard binary, but for that you could add a wee function to [the framerok](libbeat/tests/integration/framework.go) to use a regular build instead.

```go
func NewStandardBeat(t *testing.T, beatName, binary string, args ...string) *BeatProc {
	b := NewBeat(t, beatName, binary, args...)

	// remove "--systemTest"
	b.baseArgs = append(b.baseArgs[:1], b.baseArgs[2:]...)

	return b
}
```